clusterloader2/pkg/test/simpletestexecutor.go

```diff
@@ -28,20 +33,72 @@ func createSimpleTestExecutor() TestExecutor {
 
 // ExecuteTest executes test based on provided configuration.
 func (ste *simpleTestExecutor) ExecuteTest(ctx Context, conf *api.Config) error {
+	err := ctx.GetFramework().CreateAutomanagedNamespaces(int(conf.AutomanagedNamespaces))
+	defer cleanupResources(ctx)
+	if err != nil {
+		return fmt.Errorf("automanaged namespaces creation failed: %v", err)
+	}
+
+	for i := range conf.Steps {
+		ste.ExecuteStep(ctx, &conf.Steps[i])
+	}
 	return nil
 }
 
 // ExecuteStep executes single test step based on provided step configuration.
 func (ste *simpleTestExecutor) ExecuteStep(ctx Context, step *api.Step) error {
+	if len(step.Measurements) > 0 {
+		// TODO: handle measurments
+	} else {
+		var wg wait.Group
+		defer wg.Wait()
+		for i := range step.Phases {
+			phase := &step.Phases[i]
+			wg.Start(func() {
```

Comment:
Same as below - you should somehow try to capture the error from executing individual phases.